docker-compose.yml

```diff
@@ -193,6 +192,8 @@ services:
 
   iperf3:
     image: mlabbe/iperf3
+    healthcheck:
```

Comment:
This is ultimately the root cause. It looks even the server started up and accepted connections, it wasn't "fully up" for some reason.

Waiting for Docker to acknowledge that the process is actually listening fixed the issue.

A `sleep 1` before running the test or something would probably fix it too.